### PR TITLE
TIP-1239: make the config/ directory explicit

### PR DESCRIPTION
**Pull request by jjanvier**
The best to review this PR is to checkout it locally and/or to read commit by commit.

The ideas behind this PR:
- have only one location where you have to read the configuration in Community by regrouping all confs inside the `config/packages/` folder. No more `pim.yml` hidden somewhere. As a reminder, the Enterprise Edition now loads this Community `config/` folder to avoid to repeat all those files. Also, we won't need the standard editions the way we use them currently anymore.
- make explicit what's common to all envs vs what's specific for each env
- make the clear distinction between what's bundles' configuration vs services/parameters' definition

Not done here:
- clean the `config/services/` folder. I could do that in another PR if you feel it's needed.

### CODE REVIEW

File: config/packages/dev/swifmailer.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+swiftmailer:
```
Comment (nmarniesse) [inline config/packages/dev/swifmailer.yml]: Seems there is a typo in the file name

File: config/packages/test/swifmailer.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+swiftmailer:
```
Comment (nmarniesse) [inline config/packages/test/swifmailer.yml]: Same filename typo

File: config/packages/test_fake/swifmailer.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+swiftmailer:
```
Comment (nmarniesse) [inline config/packages/test_fake/swifmailer.yml]: same filename typo

File: config/packages/behat/framework.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+framework:
+    test: true
```
Comment (BitOne) [inline config/packages/behat/framework.yml]: :+1:

File: config/packages/behat/monolog.yml [hunk 5]
```diff
@@ -1,13 +1,3 @@
-doctrine:
-    orm:
-        entity_managers:
-            default:
-                metadata_cache_driver:
```
Comment (BitOne) [inline config/packages/behat/monolog.yml]: Where is this configuration now?

Comment (jjanvier) [reply config/packages/behat/monolog.yml]: @BitOne config/packages/prod/doctrine.yml

**Pull request merged**